### ISSUE TITLE
Feature request: Turbo mode

### ISSUE THREAD

**Issue by deepumukundan** [earwig/git-repo-updater#22]
Currently gitup sequentially pulls repo's one by one inside the passed in folder. What if there was a turbo mode which executes git pull on all the repo's concurrently? The tty print can be suppressed in that case and maybe a report generated (since there could be theoretically 100'ds of repo's being updated at the same time). Just a thought. Python is not my forte so can't create a PR with the suggested change :(.

**Comment by earwig** [comment]
Good idea, though we want to be careful to not flood github with 20 concurrent pulls—I don't know if that would bother them.

**Comment by deepumukundan** [comment]
I have seen this behaviour in Github Desktop, their own Git client when trying to pull from the swift repos. An activity indicator spins on the repo names in the left sidebar and completes once each update is complete. 

But agree with you if the pulls will bother them. Maybe we should not pull everything together but run batches of pulls concurrently. Anyways gitup as it is works well for me, so can't really complain :)

**Comment by mcameron** [comment]
I would prefer it to the following if it performed comparably: 

`for file in $(ls -1); do cd ${file}; git pull; cd ..; done`

Thanks for making it available anyway.

**Comment by JedMeister** [comment]
FWIW prior to using your awesome `git-repo-updater`, I had a bash script which via `xargs` updated repos 20 at a time (from GitHub) and I never had any issues. 

It made it incredibly fast (compared to doing it sequentially). I tested using more than 20 concurrent connections and whilst it did still increase the speed, the returns were diminishing so 20 seemed like the sweet spot. YMMV though...

Regardless, I never had any complaints from GitHub. IIRC at some point when I was testing larger numbers of concurrent connections, I started to get (what appeared to be random) failures. If memory serves, that was at some point above 100 concurrent connections to GH.

**Comment by brainfucksec** [comment]
> FWIW prior to using your awesome `git-repo-updater`, I had a bash script which via `xargs` updated repos 20 at a time (from GitHub) and I never had any issues.

Hi @JedMeister, can you show me this script please? I use this command:
 
`find . -name ".git" -type d | xargs -P 20 -I {} git --git-dir={} pull`

But I need the command to also print repository names, thanks in advance.
